### DISCUSSION TITLE
Does hierarchicalMenu categories.lvl0 need to be an array in react-instantsearch v7?

### DISCUSSION THREAD

**Discussion by adambremler** [algolia/instantsearch#5935] [Q&A] [answered]
In the docs [(link)](https://www.algolia.com/doc/api-reference/widgets/hierarchical-menu/react/#requirements) it seems like `categories.lvl0` can be either a string or an array of strings.

But when attempting to set the state with `categories.lvl0` as a string I get the following error: `Uncaught TypeError: values.join is not a function`. It comes from the `getWidgetSearchParameters` function in `connectHierarchicalMenu.ts`. It throws an error because `categories.lvl0` is a string and doesn't have a `.join` function. This makes it seem like `categories.lvl0` can't be a string and has to be an array.

Is this true? Appreciate any help! :)

**Comment by dhayab** [comment, marked as answer]
Hi, these are 2 different data structures.

Your dataset (your records) can indeed accept both string and array-based values, for example making one item part of many different hierarchical branches.

The InstantSearch UI state for the hierarchical menu only accepts an array of strings. Here is what a UI state composed of many widgets could look like, for example: https://www.algolia.com/doc/api-reference/widgets/ui-state/js/.

**Reply by adambremler** [reply]
Ah okay I see. Thanks a lot for taking the time to clarify this! :)
